src/base.jl

```diff
@@ -234,13 +234,14 @@ function get_components(::Type{T}, sys::ConcreteSystem) where {T <: Component}
     if isconcretetype(T)
         components = get(sys.components, T, nothing)
         if isnothing(components)
-            return Iterators.flatten([[]])
+            return Iterators.take(Vector{T}(), 0)
         else
-            return Iterators.flatten([components])
+            return Iterators.take(components, length(components))
         end
     else
         types = [x for x in get_all_concrete_subtypes(T) if haskey(sys.components, x)]
-        return Iterators.flatten(sys.components[x] for x in types)
+        abstract_type_vector = Vector{T}(reduce(vcat, sys.components[x] for x in types))
```

Comment:
Doesn't this allocate a new array? (which is what we were trying to avoid?

Reply:
It does, I guess it doesn't really help to fix the issue.

Reply:
removed with f31e366